src/components/Event/Event.tsx

```diff
@@ -26,40 +35,76 @@ export class Event extends React.Component<EventProps> {
     showTime: true,
     fullday: false,
     selected: false,
+    disabled: false,
+    onClick: null,
+    'aria-has-popup': false,
+    'aria-expanded': false,
   };
 
   getDataAttributes() {
-    const { fullday, selected, showTime } = this.props;
+    const { fullday, selected, showTime, onClick, disabled } = this.props;
 
     return {
       [EVENT_DATA_KEYS.IsFullDay]: fullday,
       [EVENT_DATA_KEYS.IsSelected]: selected,
       [EVENT_DATA_KEYS.IsTimeShown]: showTime,
+      [EVENT_DATA_KEYS.OnClick]: Boolean(onClick),
+      [EVENT_DATA_KEYS.Disabled]: disabled,
     };
   }
 
   render() {
-    const { time, title, showTime, fullday, selected, className } = this.props;
+    const {
+      time,
+      title,
+      showTime,
+      fullday,
+      selected,
+      className,
+      onClick,
+      disabled,
+    } = this.props;
     const timeComponent =
       showTime && time ? <Text className={classes.time}>{time}</Text> : null;
 
+    const eventProps = isContainer =>
+      isContainer
+        ? {
+            'data-hook': this.props['data-hook'],
+            ...this.getDataAttributes(),
+            [EVENT_DATA_KEYS.ARIA_Has_Popup]: this.props['aria-has-popup'],
+            [EVENT_DATA_KEYS.ARIA_Expanded]: this.props['aria-expanded'],
+          }
+        : null;
+
+    const event = (rtl, isContainer) => (
+      <div
+        {...eventProps(isContainer)}
+        className={st(
+          classes.root,
+          { fullday, selected, rtl, disabled },
+          className,
+        )}
+      >
+        {timeComponent}
+        <Text className={classes.title}>{title}</Text>
+        <div className={classes.overlay} />
+      </div>
+    );
```

Comment:
For readability, I prefer moving these function outside of the `render` method, and just use them as instance methods:
```js
  _getEventProps: () => {
   ...
  },
  
  _getEvent: () => {
    ...
  }
```